### PR TITLE
Test By Block

### PR DESCRIPTION
**Pull request by legaspielaine**
Added Tests by Block for all main blocks
- Block tab tests
- Style tab tests
- Advanced tab tests
- Other tests (basic block tests)

---

- [ ] Update the yml files in this PR to include running all tests (all v2 & v3)

### CODE REVIEW

File: cypress/support/gutenberg/commands/controls.js [hunk 1]
```diff
@@ -412,6 +415,100 @@ function radioControl( name, value, options = {} ) {
...
+		parentSelector,
+		supportedDelimiter,
+	} )
+
+	const isWpLessThan58 = compareVersions( Cypress.env( 'WORDPRESS_VERSION' ), '5.8.0', '<' )
+
+	beforeAdjust( name, value, options )
+	value.forEach( val => {
+		if ( isWpLessThan58 ) {
+		// The suggestions list does not display on WP versions less than 5.8. Type and enter the values.
```
Comment (blankart) [inline cypress/support/gutenberg/commands/controls.js:444]: This is false. Form token fields are designed to have suggestion list.

File: cypress/support/gutenberg/commands/controls.js [hunk 2]
```diff
@@ -412,6 +415,100 @@ function radioControl( name, value, options = {} ) {
...
+		beforeAdjust = () => {},
+		parentSelector,
+		supportedDelimiter = [],
+	} = options
+
+	const selector = name => cy
+		.getBaseControl( name, {
+			isInPopover, parentSelector, supportedDelimiter,
+		} )
+		.find( `.components-base-control__field:contains(${ name })` )
```
Comment (blankart) [inline cypress/support/gutenberg/commands/controls.js:490]: Can we use `containsRegExp` here?

File: cypress/support/gutenberg/commands/controls.js [hunk 3]
```diff
@@ -412,6 +415,100 @@ function radioControl( name, value, options = {} ) {
...
+					.find( 'input.components-checkbox-control__input' )
+					.click( { force: true } )
+			}
+		} else if ( value ) {
+			// If not yet checked and value is set to true, click the checkbox.
+			selector( name )
+				.find( 'input.components-checkbox-control__input' )
+				.click( { force: true } )
+		}
+	} )
```
Comment (blankart) [inline cypress/support/gutenberg/commands/controls.js:509]: Comparing `length > 0` to value is cleaner. ( Assuming that `length > 0` = checkbox is checked )

File: cypress/support/stackable/commands/controls.js [hunk 4]
```diff
@@ -546,3 +648,243 @@ export function adjustDesign( option = '' ) {
...
+	} = options
+
+	const selector = () => cy.getBaseControl( name, {
+		isInPopover,
+		parentSelector,
+		supportedDelimiter,
+	} )
+
+	beforeAdjust( name, value, options )
+	value.forEach( ( val, index ) => {
```
Comment (blankart) [inline cypress/support/stackable/commands/controls.js:752]: Handle `undefined` values

File: cypress/support/stackable/commands/controls.js [hunk 5]
```diff
@@ -546,3 +648,243 @@ export function adjustDesign( option = '' ) {
...
+	selectColumn()
+		.click( { force: true } )
+
+	selectColumn()
+		.find( '.stk-linking-wrapper > .stk-linking-wrapper__tooltip' )
+		.click( { force: true } )
+}
+
+/**
+ * Command for activating or deactivating the linking module in a column.
```
Comment (blankart) [inline cypress/support/stackable/commands/controls.js:857]: Update comment

File: cypress/support/stackable/commands/controls.js [hunk 6]
```diff
@@ -546,3 +648,243 @@ export function adjustDesign( option = '' ) {
...
+ *
+ * @param {string} blockName - the parent block
+ * @param {string | number | Object} selector - selector of parent block
+ * @param {Object} options
+ */
+function adjustResizableColumnWidth( blockName = 'stackable/columns', selector, options = {} ) {
```
Comment (blankart) [inline cypress/support/stackable/commands/controls.js:863]: See comments above

File: cypress/support/stackable/commands/controls.js [hunk 7]
```diff
@@ -546,3 +648,243 @@ export function adjustDesign( option = '' ) {
...
+	const {
+		index = 1, // index of the column to adjust
+		value,
+		columnDataType = 'stackable/column',
```
Comment (blankart) [inline cypress/support/stackable/commands/controls.js:867]: See comments above

**Review by blankart — CHANGES_REQUESTED** [review]

File: cypress/support/gutenberg/commands/controls.js (hunk 1 above)
Comment (legaspielaine) [reply cypress/support/gutenberg/commands/controls.js:444]: I meant when the input is initially clicked. `__experimentalExpandOnFocus` only works on WP 5.8

Comment (legaspielaine) [reply cypress/support/gutenberg/commands/controls.js:444]: will update the comment

File: cypress/support/stackable/commands/controls.js [hunk 8]
```diff
@@ -508,7 +526,91 @@ function iconControlReset( name, options = {} ) {
...
+		.click( { force: true } )
+}
+
+/**
+ * Command for resetting the dynamic content control.
+ *
+ * @param {string} name
+ * @param {Object} options
+ */
+function dynamicContentControlReset( name, options = {} ) {
```
[the remark this answers is not in the document]
Comment (legaspielaine) [reply cypress/support/stackable/commands/controls.js:539]: This is injected in resetStyle

File: cypress/support/stackable/commands/controls.js [hunk 9]
```diff
@@ -546,3 +648,243 @@ export function adjustDesign( option = '' ) {
...
+ * Command for activating or deactivating the linking module in a column.
+ *
+ * @param {string} blockName - the parent block
+ * @param {string | number | Object} selector - selector of parent block
+ * @param {Object} options
+ */
+function activateOrDeactivateLinking( blockName = 'stackable/columns', selector, options = {} ) {
+	const {
+		index = 1, // index of the column to adjust
+		columnDataType = 'stackable/column',
```
[the remark this answers is not in the document]
Comment (legaspielaine) [reply cypress/support/stackable/commands/controls.js:839]: try chanining the command: previous subj true

File: cypress/support/stackable/commands/controls.js (hunk 6 above)
Comment (legaspielaine) [reply cypress/support/stackable/commands/controls.js:863]: rename to resizeWidth, chainable

File: cypress/support/gutenberg/commands/controls.js [hunk 10]
```diff
@@ -412,6 +434,141 @@ function radioControl( name, value, options = {} ) {
...
+		beforeAdjust = () => {},
+		parentSelector,
+		supportedDelimiter = [],
+	} = options
+
+	const selector = name => cy
+		.getBaseControl( name, {
+			isInPopover, parentSelector, supportedDelimiter,
+		} )
+		.find( `.components-base-control__field:contains(${ containsRegExp( name ) })` )
```
Comment (blankart) [inline cypress/support/gutenberg/commands/controls.js:502]:
```suggestion
		.find( '.components-base-control__field' )
		.contains( containsRegExp( name )  )
```

**Review by blankart — APPROVED** [review]

File: cypress/support/gutenberg/commands/controls.js (hunk 2 above)
Comment (legaspielaine) [reply cypress/support/gutenberg/commands/controls.js:526]: tests are failing if we use `.contains( containsRegExp( name ) )`. We need to yield `.components-base-control__field`

File: .github/workflows/e2e-test-wp55.yml [hunk 11]
```diff
@@ -3,7 +3,7 @@ on:
   push:
     branches: [ master ]
   pull_request:
-    branches: [ master ]
+    branches: [ master, v3-commands ]
```
Comment (blankart) [inline .github/workflows/e2e-test-wp55.yml:6]:
```suggestion
    branches: [ master ]
```

**Pull request merged**